### PR TITLE
Understanding how cluster markers work

### LINKED ISSUE #4841: Understanding how cluster markers work in web mapping using leaflet.js
#### Proposed title of article
[Web-GIS] Understanding how cluster markers work in web mapping using leaflet.js

#### Proposed article introduction
Understanding how cluster markers work in leaflet.js

As we all know Big Data is the keyword for the 21st century. A few people realize that a substantial part of these massive quantities of data is categorized as geospatial. Nowadays almost all the devices we used are satellite-based radio navigation systems like GPS.

Nonetheless even today, the simplest case of displaying hundreds of Points on a web map with acceptable usability can be a real challenge. Especially from the perspective of the user, the navigation experience can be utterly painful. For instance, [smartrak](http://www.smartrak.co.nz/), this platform has customers with thousands of points on the map. When you zoom it out, these markers all overlap and make the map looks messy and crowded.

To improve this, it is professional to use marker clustering, a technique of grouping markers that are close to each other together on each zoom level.

#### Key takeaways 
-Getting started with leaflet.js map
-Animated map interaction
-Group markers into clusters

#### Article quality
This article will help understand how to group markers close to each other and improve the look and interaction of websites like [smartrak](http://www.smartrak.co.nz/), 

#### References
[Leaflet.js](https://leafletjs.com)
[Marker Cluster](https://leafletjs.com/2012/08/20/guest-post-markerclusterer-0-1-released.html)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Adebogunabdulroheem**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4841

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers

Is anyone available to review this PR?

Thank you

**Comment by jumamiller** [comment]
> @section-engineering-education/peer-reviewers
> 
> Is anyone available to review this PR?
> 
> Thank you

@hectorkambow I'm available to work on this!

File: content/articles/understanding-how-cluster-marker-works/index.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+### Understanding how cluster markers work in web mapping using leaflet.js
+  This is no more a news, as we are all aware that big data is the keyword of the 21st century, and It's shocking to realize that only a few people understand that a large part of data is categorized as geospatial. Nowadays, nearly all of the gadgets we use are satellite-based systems like GPS. Today, the simplest case of displaying data/marker on a map with an acceptable user experience may be a real challenge especially for the user as they might find it difficult to understand and to analyze data at a particular zoom level. For instance, a platform that has users with hundreds of markers on the map, when you zoom out the map, these markers all overlap and make the map appear messy and crowdy. To improve this kind of map, it is recommended to use the marker clustering method.
```
Comment (jumamiller) [inline content/articles/understanding-how-cluster-marker-works/index.md:2]: This is a very hard-to-read paragraph. Break it down into multiple paragraphs.

File: content/articles/understanding-how-cluster-marker-works/index.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+### Prerequisites
+To Understand this tutorial, it is assumed that the reader knows the below technologies:
```
Comment (jumamiller) [inline content/articles/understanding-how-cluster-marker-works/index.md:5]: Work on this grammar. Go straight to the point and make this article flow!

File: content/articles/understanding-how-cluster-marker-works/index.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+- JavaScript
+- HTML/CSS
+  
+### Goal
+At the end of this guide, we'll build a map with clustering features, and these include:
```
Comment (jumamiller) [inline content/articles/understanding-how-cluster-marker-works/index.md:10]: You don't build the features at the end!! Do you build it from the start? In the end what should have the learner gain reading your article?

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, the article is of very low quality, in terms of grammar and flow and even the formatting. Kindly work on it afresh to avoid its closure! Thank you!

**Comment by jumamiller** [comment]
Hi, I'll close this for now, kindly re-submit a ready-to-publish article!

**Pull request closed without merging**